warehouse/views.py

```diff
@@ -326,11 +326,41 @@ def filter_key(item):
         except ValueError:
             return 1, 0, item[0]
 
+    def form_filters_tree(split_list):
+        """
+        Takes a list of lists each of them containing filter and
+        one of it's child.
+        Returns dictionary with key being filter and value being
+        it's children
+        """
+        d = {}
+        for l in split_list:
+            current_level = d
+            for part in l:
+                if part not in current_level:
+                    current_level[part] = {}
+                current_level = current_level[part]
+        return d
+
+    def process_available_filters():
+        """
+        Processes available filters and returns a list of dictionaries
+        Value of a key in the dictionary represents it's children
+        """
```

Comment:
Better grammar (assuming that my understanding is accurate):

Processes available filters and returns a list of dictionaries.
The value of a key in the dictionary represents its children.